csrc/device_lower/pass/replace_size.cpp

```diff
@@ -52,18 +52,40 @@ std::unordered_map<Val*, Val*> getSimplificationMap(Fusion* fusion) {
 
   std::unordered_map<Val*, Val*> simplification_map;
 
+  auto get_number_of_defs = [](Val* val) -> int64_t {
+    std::deque<Val*> vals;
+    vals.push_back(val);
+    int64_t num_defs = 0;
+    while (!vals.empty()) {
+      auto v = vals.front();
+      vals.pop_front();
+      auto def = v->definition();
+      if (def == nullptr) {
+        continue;
+      }
+      ++num_defs;
+      for (auto inp : def->inputs()) {
+        vals.push_back(inp);
+      }
+    }
+    return num_defs;
+  };
+
   for (const ValGroup& group : graph.disjointValSets().disjointSets()) {
     // For each ValGroup, find a single extent to use for all extents of
     // IterDomains in the group. These are chosen in descending order of
     // preference:
     // 1. Constant ints. These might be non-immediate constants
     // 2. Extents of input TVs.
     // 3. Extents of non-input TVs.
-    // Within these three classes, we find the IterDomain with the smallest
-    // name().
+    // Within these three classes, we find the IterDomain with the
+    // smallest name(). For case 3, we also prefer the IterDomain with
```

Comment:
I guess we could also do this for Case 1 just for readability, but it shouldn't affect the compiled kernel so not important..

Reply:
Yeah, I didn't do that just because `get_number_of_defs` may not be cheap.